### ISSUE TITLE
Mac IDE: Shift-Option-ArrowKeys not behaving as expected during text entry.

### ISSUE THREAD

**Issue by PrismaticRealms** [YoYoGames/GameMaker-Bugs#2438]
### Description
Throughout the Gamemaker IDE, wherever text is entered (i.e. code editor, renaming assets, etc.) the macOS keyboard shortcuts Shift-Option-LeftArrow and Shift-Option-RightArrow should be extending the text selection while moving the insertion point to the beginning or end of the current or next word, but it does not.  Instead it only moves the insertion point by a single character and extending the selected text by that one character.  Note that hitting those key combinations WITHOUT holding down shift DOES move the insertion point whole words at a time.  Holding Shift while doing so should do the same but should also extend the text selection as well.

This behaviour seems to work fine on the PC GameMaker IDE (using Shift-Ctrl-Arrows instead.)

### Steps To Reproduce
On the Mac IDE, open the code editor, type in (or paste in) some text consisting of multiple words.  Now try the aformentioned keyboard combinations to try to select whole words at a time.  Observe that only a single character is selected each time the key combos are hit instead of entire words.

[c98844e9-31e9-478a-910f-fd8496f9ae2e](https://api.gamemaker.io/api/github/downloads/c98844e9-31e9-478a-910f-fd8496f9ae2e)

**Comment by PrismaticRealms** [comment]
Note: Shift-Command-Arrow keys to select text to end-of-line/start-of-line works fine.

**Comment by PrismaticRealms** [comment]
Related bug: When selecting text using Shift-Option-Arrow Left/Right, copying that text and then pasting that text elsewhere, the insertion point should end up at the end of the text being pasted, but instead the insertion point ends up at the start of the text being pasted.

**Comment by SmithTom6304** [comment]
Windows `Caret Left + Column Select` and `Caret Right + Column Select` hotkeys were conflicting with the Mac `Move Token Left + Select` and `Move Token Right + Select` hotkeys.

I've also unbound the Windows `Caret Up + Column Select` and `Caret Down + Column Select` hotkeys (by default `Alt, Shift, Up/Down`) and instead added those hotkeys as an alternate to `Home + Select` and `End + Select`. I've then added the non-select variants `Alt, Up/Down` as an alternate to `Home` and `End`. This matches the functionality seen in other text editing apps.

So in summary:
- Alt Left/Right moves cursor to the previous/next token
- Alt Up/Down moves cursor to the start/end of the line (same behaviour as Home/End)
- Holding shift during these operations will select text

Note that these changes apply to Mac only.

**Comment by YYBartT** [comment]
Updated the manual's keyboard shortcuts page, no changes to the key combinations mentioned here as they are considered unlisted.

**Comment by alicemoretti** [comment]
Verified fixed on IDE v2024.2.0.131  Runtime v2024.2.0.162.

Thank you.
